app/components/Home.tsx

```diff
@@ -0,0 +1,368 @@
+      tokens: any,
+      idx: any,
+      options: any,
+      env: any,
+      self: any
+    ) {
+      // If you are sure other plugins can't add `target` - drop check below
+      let aIndex = tokens[idx].attrIndex('target');
+
+      if (aIndex < 0) {
+        tokens[idx].attrPush(['target', '_blank']); // add new attribute
+      } else {
+        tokens[idx].attrs[aIndex][1] = '_blank'; // replace value of existing attr
+      }
+
+      // pass token to default renderer.
+      return defaultRender(tokens, idx, options, env, self);
+    };
+  };
+
+  updatePreviewText = () => {
+    const preview = document.getElementById(HtmlElementId.Preview);
+    if (preview) {
+      preview.innerHTML = MarkdownParser.render(this.state.text);
+    }
+  };
+
+  removeSelection = () => {
+    if (window.getSelection) {
+      window.getSelection().removeAllRanges();
+    }
+  };
+
+  configureResizer = () => {
+    const simpleMarkdown = document.getElementById(
+      HtmlElementId.SimpleMarkdown
+    );
+    const editor = document.getElementById(HtmlElementId.Editor);
+    const columnResizer = document.getElementById(HtmlElementId.ColumnResizer);
+    let pressed = false;
+    let startWidth = editor.offsetWidth;
+    let startX;
+    let lastDownX;
+
+    let resizerWidth = columnResizer.offsetWidth;
+    let safetyOffset = 15;
+
+    columnResizer.addEventListener(MouseEvent.Down, (event) => {
+      pressed = true;
+      lastDownX = event.clientX;
+      columnResizer.classList.add('dragging');
+      editor.classList.add('no-selection');
+    });
+
+    document.addEventListener(MouseEvent.Move, (event) => {
+      if (!pressed) {
+        return;
+      }
+
+      let x = event.clientX;
+      if (x < resizerWidth / 2 + safetyOffset) {
+        x = resizerWidth / 2 + safetyOffset;
+      } else if (x > simpleMarkdown.offsetWidth - resizerWidth - safetyOffset) {
+        x = simpleMarkdown.offsetWidth - resizerWidth - safetyOffset;
+      }
+
+      let colLeft = x - resizerWidth / 2;
+      columnResizer.style.left = colLeft + 'px';
+      editor.style.width = colLeft - safetyOffset + 'px';
+
+      this.removeSelection();
+    });
+
+    document.addEventListener(MouseEvent.Up, (event) => {
+      if (pressed) {
+        pressed = false;
+        columnResizer.classList.remove('dragging');
+        editor.classList.remove('no-selection');
+      }
+    });
+    // At some point we want to remove the EventListers
+  };
+
+  onKeyDown = (event: React.KeyboardEvent) => {
+    keyMap.set(event.key, true);
+    if (!keyMap.get('Shift') && keyMap.get('Tab')) {
+      event.preventDefault();
+      document.execCommand('insertText', false, '\t');
+    }
+  };
+
+  onKeyUp = (event: React.KeyboardEvent) => {
+    keyMap.delete(event.key);
+  };
+
+  onBlur = () => {
+    keyMap.clear();
+  };
+
+  logDebugMessage = (message: string, object: any) => {
+    if (debugMode) {
+      console.log(message + object);
```

Comment:
Should be `console.log(message, object);`

Reply:
OK 👍